File: Go/find_index_of_the1st_occurrence_in_a_string/28_test.go
package findindexofthe1stoccurrenceinastring

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func Test_strStr(t *testing.T) {
	tests := []struct {
		haystack string
		needle   string
		want     int
	}{
		{haystack: "sadbutsad", needle: "sad", want: 0},
		{haystack: "leetcode", needle: "leeto", want: -1},
		{haystack: "a", needle: "a", want: 0},
		{haystack: "hello", needle: "ll", want: 2},
		{haystack: "abbb", needle: "aaaa", want: -1},
	}

	for i, tt := range tests {
		t.Run(fmt.Sprintf("case %d", i+1), func(t *testing.T) {
			got := strStr(tt.haystack, tt.needle)
			assert.Equal(t, tt.want, got)
		})
	}
}

File: TypeScript/find_index_of_first_occur/28.test.ts
import strStr from './28';
import { describe, test, expect } from 'bun:test';

interface tt {
  haystack: string;
  needle: string;
  want: number;
}

const tests: tt[] = [
  { haystack: 'sadbutsad', needle: 'sad', want: 0 },
  { haystack: 'leetcode', needle: 'leeto', want: -1 },
  { haystack: 'hello', needle: 'll', want: 2 },
  { haystack: 'a', needle: 'a', want: 0 },
  { haystack: 'abc', needle: 'c', want: 2 },
  { haystack: 'baaa', needle: 'aaaa', want: -1 },
];

describe('find index of the first occur string', () => {
  tests.forEach((t, i) => {
    test(`case ${i + 1}`, () => {
      const result = strStr(t.haystack, t.needle);
      expect(result).toBe(t.want);
    });
  });
});
